### ISSUE TITLE
Add a button to delete Source Definitions and Destination Definitions

### ISSUE THREAD

**Issue by pmossman** [airbytehq/airbyte#9097]
## Tell us about the problem you're trying to solve
The API now supports deleting source definitions and destination definitions.

- The source definition deletion endpoint spec can be found here: https://github.com/airbytehq/airbyte/blob/master/airbyte-api/src/main/openapi/config.yaml#L317-L335
- The destination definition deletion endpoint spec can be found here: https://github.com/airbytehq/airbyte/blob/master/airbyte-api/src/main/openapi/config.yaml#L652-L670

The UI should allow the user to delete source definitions and destination definitions in much the same way that users can delete source and destination connectors.

## Describe the solution you’d like
In the `settings/source` and `settings/destination` page, each row in the table should include a red `delete` button. When clicked, a confirmation modal should appear asking the user to confirm removal. The modal should include a warning:

> This cannot be undone. Note that:
> - All connectors and connections based on this _(source or destination)_ will also be deleted, including past logs and configurations. 
> - No existing data in affected sources and destinations will be altered.

The modal footer should include a cancel button and a delete button, just like the modal for deletions of sources/destinations/connections.

## Additional context
The original issue for adding the API endpoint is here: https://github.com/airbytehq/airbyte/issues/8732
It was determined that this chance is small enough to not need mockups, but please let me know if anything needs further clarification!

**Priority: Medium**

**Comment by timroes** [comment]
@pmossman Do we have any specific permissions required to do that? Is this generally disabled on Cloud, or e.g. how do we determine that users can't do that on Cloud?

**Comment by timroes** [comment]
@pmossman I wonder if we might want to still not show the button even for Admins on Cloud? Just to prevent any cloud admin (airbyte employee) to accidentally delete a connector definition in prod cloud?

**Comment by pmossman** [comment]
Yeah I think the original intent of the endpoint was primarily for being able to delete custom connector definitions, which aren't currently available in Cloud at all. I think hiding it in Cloud for now probably makes a lot of sense
